stack/core/src/main/java/org/apache/usergrid/persistence/PathQuery.java

```diff
@@ -122,10 +126,30 @@ protected Results getHeadResults( EntityManager em ) throws Exception {
 
         EntityRef ref = new SimpleEntityRef(type,uuid);
 
-        // if it's a single name identifier, just directly fetch that
-        if ( !query.getQl().isPresent() && query.getSingleNameOrEmailIdentifier() != null){
+        if ( !query.getQl().isPresent() && query.getSingleUuidIdentifier() != null) {
```

Comment:
What is the reasoning behind this change?

Reply:
I believe this was because push notification devices being specified by UUID were being returned using ES instead of direct access.